### PR TITLE
Further adapt to OTP 24 while bumping some dep.s

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
I'll still test this as a `_checkout` in `elvis` and `rebar3_lint`, but this is a move forward.

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
Still working out some warnings on OTP 24.

**Comment by paulo-ferraz-oliveira** [comment]
So here's a difference:
* OTP 23
```erlang
#{ attrs => #{
       location => {23, 5},
       name => "module_name",
       text => "?module_name"
   },
   content => [],
   type => macro }
```
* OTP 24
```erlang
#{ attrs => #{
       location => {23, 5},
       text => "?"
   },
   content => [],
   node_attrs => #{
       function=> #{
           attrs => #{
               location => {23, 5},
               text => "?"
           },
           node_attrs => #{
               function => #{
                   attrs => #{
                       location => {23, 18},
                       name => "function_name",
                       text => "?function_name"
                   },
                   content => [],
                   type => macro
               },
               module => #{
                   attrs => #{
                       location => {23, 5},
                       name => "module_name",
                       text => "?module_name"
                   },
                   content => [],
                   type => macro
               }
           },
           type => remote
       }
   },
   type => call
 }
```

(that's always fun)

**Comment by paulo-ferraz-oliveira** [comment]
@elbrujohalcon, @jfacorro ^ does that seem more `elvis_core`-related or `katana-code`-related?

**Comment by paulo-ferraz-oliveira** [comment]
I'm thinking more `elvis_core`, since this is interpreted different: `?module_name:?function_name()`.
In OTP 23 it's a macro followed by stuff, in OTP 24 it's a function.

File: elvis.config [hunk 1]
```diff
@@ -18,10 +18,6 @@
                   {elvis_style, atom_naming_convention, #{ignore => [elvis_task]}}],
         ruleset => beam_files
        },
-      #{dirs => ["."],
```
Comment (paulo-ferraz-oliveira) [inline elvis.config:21]: Unused, so not need for the lib. itself. (also, `elvis_core` warns about this 😄).

File: rebar.config [hunk 2]
```diff
@@ -23,11 +23,11 @@
...
     {extra_src_dirs, ["test/examples"]},
-    {deps, [ {mixer,       "1.0.1", {pkg, inaka_mixer}}
-           , {meck,        "0.9.0"}
-           , {xref_runner, "1.1.1"}
+    {deps, [ {mixer,       "1.1.1", {pkg, inaka_mixer}}
+           , {meck,        "0.9.2"}
+           , {xref_runner, "1.2.0"}
     ]},
-    {erl_opts, [ nowarn_missing_spec ]},
+    {erl_opts, [ nowarn_missing_spec, nowarn_export_all ]},
```
Comment (paulo-ferraz-oliveira) [inline rebar.config:30]: This avoid warnings, but in reality fixes no real issue apart from that.

File: test/style_SUITE.erl [hunk 3]
```diff
@@ -163,7 +163,7 @@ verify_function_naming_convention(Config) ->
     PathFail = "fail_function_naming_convention." ++ Ext,
 
     [_CamelCaseError, _ALL_CAPSError, _InitialCapError,
-     _HyphenError, _PredError, _EmailError] =
```
Comment (paulo-ferraz-oliveira) [inline test/style_SUITE.erl:166]: This is OTP 24 complaining, and it's right 😄.

File: src/elvis_style.erl [hunk 4]
```diff
@@ -953,6 +953,8 @@ has_remote_call_parent(Zipper) ->
     case ktn_code:type(Node) of
         remote ->
             true;
+        call ->
```
Comment (paulo-ferraz-oliveira) [inline src/elvis_style.erl:956]: I started this with a `has_remote_call_parent(zipper:down(Zipper))` implementation, but was afraid we could fall into infinite recursion `up`/`down`/`up`/`down`/`up`/`down`/... (this fixes the tests for OTP 24 and also runs on previous versions, but I'm not exactly 💯 on this change).

File: test/style_SUITE.erl (hunk 3 above)
Comment (elbrujohalcon) [reply test/style_SUITE.erl:166]: I think the original idea was not for those two errors to match, just to ignore both.
In other words, I would've replaced the other one by `_EmailError2` or something like that.

**Review by elbrujohalcon — APPROVED** [review]
No idea about the `zipper` parts… but the PR is fine in general.

File: test/style_SUITE.erl (hunk 3 above)
Comment (paulo-ferraz-oliveira) [reply test/style_SUITE.erl:166]: Sure, I can do that (which is why I think the warning is very nice), but it's working in any case 😄

Comment (paulo-ferraz-oliveira) [reply test/style_SUITE.erl]: 23709ab

**Pull request merged**